enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,432 @@
+    ExternalPlatformType PlatformType = "External"
+	...
+)
+```
+
+Additionally, respective external platform spec and status should be added to the infrastructure resource. 
+
+```go
+type CloudControllerManagerMode string
+
+const (
+    // Cloud Controller Manager is enabled and expected to be supplied.
+	// Signaling MCO to set `--cloud-provider=external` flag to the kubelets.
+    CloudControllerManagerEnabled CloudControllerManagerMode = "Enabled"
+    // No Cloud Controller Manager is expected to be supplied.
+    // Signaling MCO not to set `--cloud-provider` flag to the kubelets.
+    CloudControllerManagerDisabled CloudControllerManagerMode = "Disabled"
+)
+
+// ExternalPlatformSpec holds the desired state for the generic External infrastructure provider.
+type ExternalPlatformSpec struct{
+    // CloudControllerManager determines whether or not an external Cloud Controller Manager is expected to 
+    // be presented in the cluster.
+    // For engaging an external Cloud Controller Manager, certain flags are expected to be set to the kubelets.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When enabled, the respective operator (machine config operator) should set `--cloud-provider=external` flag to the kubelet.
+    // When omitted or disabled, no `cloud-provider` flag should be set.
+    // +kubebuilder:validation:Enum=Enabled;Disabled
+    // +optional
+    CloudControllerManager CloudControllerManagerMode `json:"cloudControllerManager,omitempty"`
+}
+
+// ExternalPlatformStatus holds the current status of the generic External infrastructure provider.
+type ExternalPlatformStatus struct{}
+```
+
+```go
+type PlatformSpec struct {
+    ...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformSpec `json:"external,omitempty"`
+}
+
+type PlatformStatus struct {
+	...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformStatus `json:"external,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBD
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create support burden. However, using of such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signals about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle hardened.
+Changing it with adding new, way less specific platform type would mean a significant step away from current design patterns in this area.
```

Comment:
my question here would be, what is the repercussion of this step away from our current designs?